### PR TITLE
[connectedmachine] release stable version 2022-12-27

### PR DESCRIPTION
**Pull request by yaotongms**
This release is targeting version 2022-12-27 of connectedmachine and using aaz. You will see all commands from the previous version with the following new commands:

- az connectedmachine extension image list: display metadata of all extensions
- az connectedmachine extension image show: display metadata of extensions with specific version
- az connectedmachine install-patches: install patches on an azure-enabled server
- az connectedmachine assess-patches: assess patches on an azure-enabled server

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yaotongms** [comment]
the corresponding aaz PR: https://github.com/Azure/aaz/pull/227

**Comment by yaotongms** [comment]
> @microsoft-github-policy-service agree company="Microsoft"

@microsoft-github-policy-service agree company="Microsoft"

**Comment by yaotongms** [comment]
> @yaotongms please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement



> @yaotongms please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

File: src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py [hunk 1]
```diff
@@ -21,124 +21,134 @@
...
+                '--name "{machine}" '
+                '--maximum-duration "PT4H" '
+                '--reboot-setting "IfRequired" '
+                '--windows-parameters "{{\\"classificationsToInclude\\":[\\"Critical\\", \\"Security\\"]}}"',
+                checks=[
+                    self.check('properties.resourceGroup', '{rg}'),
+                    self.check('properties.name', '{machine}'),
+                    self.check('properties.maximumDuration', 'PT4H'),
+                    self.check('properties.rebootSetting', 'IfRequired')
+        ])
```
Comment (zhoxing-ms) [inline src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py:79]: > > > except CannotOverwriteExistingCassetteException as ex:
>           raise AssertionError(ex)
E           AssertionError: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/connectedmachine/azext_connectedmachine/tests/latest/recordings/test_machine_and_extension.yaml') in your current record mode ('once').
E           No match for the request (<Request (POST) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/installPatches?api-version=2022-12-27>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/installPatches?api-version=2022-12-27%3E)) was found.
E           Found 1 similar requests with 1 different matcher(s) :
E           
E           1 - (<Request (POST) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/upgradeExtensions?api-version=2022-12-27>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/upgradeExtensions?api-version=2022-12-27%3E).)
E           Matchers succeeded : ['method', 'scheme', 'host', 'port', '_custom_request_query_matcher']
E           Matchers failed :
E           path - assertion failure :
E           /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/installPatches != /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/upgradeExtensions

Comment (zhoxing-ms) [reply src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py:79]: The yaml file of `test_machine_and_extension` did not record the `installPatches` request for the `az connectedmachine install-patches` command. Please check if the REST call in the `az connectedmachine install-patches` command was triggered when recording the yaml file in your local environment.

**Comment by zhoxing-ms** [comment]
Please resolve the Static Analysis issues and CLI Linter issues

File: src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py (hunk 1 above)
Comment (yaotongms) [reply src/connectedmachine/azext_connectedmachine/tests/latest/test_ConnectedMachineAndExtension_scenario.py]: > > except CannotOverwriteExistingCassetteException as ex:
> > raise AssertionError(ex)
> > E           AssertionError: Can't overwrite existing cassette ('/mnt/vss/_work/1/s/src/connectedmachine/azext_connectedmachine/tests/latest/recordings/test_machine_and_extension.yaml') in your current record mode ('once').
> > E           No match for the request (<Request (POST) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/installPatches?api-version=2022-12-27>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/installPatches?api-version=2022-12-27%3E)) was found.
> > E           Found 1 similar requests with 1 different matcher(s) :
> > E
> > E           1 - (<Request (POST) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/upgradeExtensions?api-version=2022-12-27>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/upgradeExtensions?api-version=2022-12-27%3E).)
> > E           Matchers succeeded : ['method', 'scheme', 'host', 'port', '_custom_request_query_matcher']
> > E           Matchers failed :
> > E           path - assertion failure :
> > E           /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/installPatches != /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/az-sdk-test/providers/Microsoft.HybridCompute/machines/testmachine/upgradeExtensions

added on the latest commit

**Comment by yaotongms** [comment]
> Please resolve the [Static Analysis issues](https://github.com/Azure/azure-cli-extensions/pull/6522/checks?check_run_id=15343324754) and [CLI Linter issues](https://github.com/Azure/azure-cli-extensions/pull/6522/checks?check_run_id=15343122981)

fixed in the latest commit

**Comment by yanzhudd** [comment]
could you please add some description messages for this PR in https://github.com/Azure/azure-cli-extensions/pull/6522#issue-1808545448?

File: src/connectedmachine/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 Release History
 ===============
+0.6.0
++++++	
+* Adding install-patches/assess-patches/Extensionimage API's. 2022/12/27 is used for aaz generation. Migrated to aaz.
```
Comment (yanzhudd) [inline src/connectedmachine/HISTORY.rst:7]:
```suggestion
* Add install-patches/assess-patches/Extensionimage API's. 2022/12/27 is used for aaz generation. Migrated to aaz.
```

**Comment by yaotongms** [comment]
> could you please add some description messages for this PR in [#6522 (comment)](https://github.com/Azure/azure-cli-extensions/pull/6522#issue-1808545448)?

description added at the top

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/_delete.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "connectedmachine delete",
+    confirmation="Are you sure you want to perform this operation?",
+)
+class Delete(AAZCommand):
+    """Delete operation to delete an Azure Arc-Enabled Server.
```
Comment (yanzhudd) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/_delete.py:19]: please keep consistent with the help messages of other commands to use verb as the beginning of the message.
```suggestion
    """Delete an Azure Arc-Enabled Server.
```

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_delete.py [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "connectedmachine extension delete",
+    confirmation="Are you sure you want to perform this operation?",
+)
+class Delete(AAZCommand):
+    """The operation to delete the extension.
```
Comment (yanzhudd) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_delete.py:19]:
```suggestion
    """Delete the extension.
```

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_list.py [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "connectedmachine extension list",
+)
+class List(AAZCommand):
+    """The operation to get all extensions of a Non-Azure machine.
```
Comment (yanzhudd) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_list.py:18]:
```suggestion
    """Get all extensions of a Non-Azure machine.
```

File: src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_show.py [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+@register_command(
+    "connectedmachine extension show",
+)
+class Show(AAZCommand):
+    """The operation to create or update the extension.
```
Comment (yanzhudd) [inline src/connectedmachine/azext_connectedmachine/aaz/latest/connectedmachine/extension/_show.py:18]:
```suggestion
    """Create or update the extension.
```

**Comment by yanzhudd** [comment]
> ❌Azure CLI Extensions Breaking Change Test
Please verify if there is any breaking change in this PR.

**Comment by yaotongms** [comment]
> connectedmachine private-link-scope update

The breaking changes include changing in parameter names and removing commands that users won't be able to use (calling internal apis). This release should not contain any breaking change.

**Review by yanzhudd — APPROVED** [review]

File: src/connectedmachine/setup.py [hunk 7]
```diff
@@ -10,7 +10,7 @@
 from setuptools import setup, find_packages
 
 # HISTORY.rst entry.
-VERSION = '0.5.1'
```
Comment (zhoxing-ms) [inline src/connectedmachine/setup.py:13]: In fact, we only recommend upgrading to the major version when there is a breaking change. If this PR does not include the breaking change, could we only upgrade to the secondary version?

File: src/connectedmachine/setup.py [hunk 8]
```diff
@@ -10,7 +10,7 @@
...
+VERSION = '1.0.0'
```
Comment (zhoxing-ms) [inline src/connectedmachine/setup.py:13]:
```suggestion
VERSION = '0.6.0'
```

File: src/connectedmachine/setup.py (hunk 7 above)
Comment (yaotongms) [reply src/connectedmachine/setup.py:13]: Updated version number to 0.6.0

**Pull request merged**